src/window.rs

```diff
@@ -60,16 +58,46 @@ impl Window {
             fb
         };
 
-        if !self.crtc_set {
-            self.crtc_set = true;
+        if self.frame_count == 0 {
+            // NOTE(mbernat): It's possible to avoid initial mode setting
+            // since we're keeping the previous mode: we could just call page_flip directly.
             self.drm_display.set_mode_with_framebuffer(Some(fb));
         } else {
             self.drm_display.page_flip(fb);
         }
+
+        // NOTE(mbernat): This buffer object returns to the surface's queue upon destruction.
+        // If we were to release it here, it would be again available from
+        // `Surface::lock_front_buffer()` next and the app would be effectively single-buffered.
+        // So, we keep it around for one frame, which is fine for double buffering.
+        self.previous_bo.replace(buffer_object);
+
+        self.frame_count += 1;
     }
 
     pub fn restore_original_display(&self) {
-        self.drm_display.set_mode_with_framebuffer(self.drm_display.crtc.framebuffer());
+        let handle = self.drm_display.crtc.framebuffer().unwrap();
```

Comment:
```suggestion
        let handle = self.drm_display.crtc.framebuffer().expect("Window should have a CRTC framebuffer handle");
```

Reply:
I postponed all error handling to #6 but it's a terrible approach, I really should be making nice errors as I go, at least with `expect`...